### ISSUE TITLE
Problem with localforage.INDEXEDDB driver on Chrome v42?

### ISSUE THREAD

**Issue by xlozinguez** [mozilla/localForage#377]
Is there a reason why using:
```
$localForageProvider.config({
            driver      : localforage.INDEXEDDB, // if you want to force a driver
            name        : 'myDb', 
```
would cause the error `No available storage method found.`

If I don't specify anything, it automatically falls back to WebSQL... I'm using the latest Chrome Version 42.0.2311.135  and according to http://caniuse.com/#search=indexeddb it should work :/

Help! :)

**Comment by thgreasi** [comment]
[This pen](http://codepen.io/thgreasi/pen/ZGWeje) works for me on chrome 42.

**Comment by xlozinguez** [comment]
You are right @thgreasi it works for me too :sob: I guess it's my app setup then...

**Comment by xlozinguez** [comment]
@thgreasi did you try to run your example with the Chrome Developer Tool open? Amazing stuff happens then! Any idea why it does this?

**Comment by xlozinguez** [comment]
@thgreasi check my screencast http://screencast.com/t/NYtCFs9d3T

**Comment by thgreasi** [comment]
Yes I tried it with the DevTools open.
Could not reproduce what your video shows.

**Comment by 3melectronicmonitoring** [comment]
I'm using localForage in a cordova app and when I debug it with chrome & dev tools open (in device mode), with an iOS device selected, I get this error.

I have localForage configured to use a SQLite driver first and then fall back on IndexedDB. If you don't support IndexedDB for safari that would make sense but I don't see any documentation stating that being the case. The dev tools will change the user agent string value when you switch between different devices to emulate and I'm assuming that's how this decision is being made.

**Comment by thgreasi** [comment]
@3melectronicmonitoring This might be related to #254 where we prevent IndexedDB for iOS 8, since that's the only sniffing that we are using.
Can you elaborate on the error?

**Comment by williamle8300** [comment]
The problem is actually quite sneaky.

TLDR; Apple's UIWebView doesn't support indexedDB.

What's happening is that Chrome spoofs the iOS user-agent when you emulate an iPhone. Well, this spoofing is actually really effective because `localForage` will now emit an error if you configure it to exclusively use indexedDB.

If you want to have a moment of profound ennui and see indexedDB in action... just turn off Chrome dev tool's emulation so that you're actually using Chrome's own user-agent. It should work then.

You can read more info here: https://github.com/mozilla/localForage/issues/254. TLDR;x2 indexedDB for iOS is not ready.

**Comment by xlozinguez** [comment]
Thanks for that response @williamle8300 , finally it makes sense! I realize recently how much the chrome dev tool actually do to emulate the iPhone interaction. Your answer is awesome, thank you.

Xavier mobile version

**Comment by tuefekci** [comment]
> The problem is actually quite sneaky.
> 
> TLDR; Apple's UIWebView doesn't support indexedDB.
> 
> What's happening is that Chrome spoofs the iOS user-agent when you emulate an iPhone. Well, this spoofing is actually really effective because `localForage` will now emit an error if you configure it to exclusively use indexedDB.
> 
> If you want to have a moment of profound ennui and see indexedDB in action... just turn off Chrome dev tool's emulation so that you're actually using Chrome's own user-agent. It should work then.
> 
> You can read more info here: #254. TLDR;x2 indexedDB for iOS is not ready.



> The problem is actually quite sneaky.
> 
> TLDR; Apple's UIWebView doesn't support indexedDB.
> 
> What's happening is that Chrome spoofs the iOS user-agent when you emulate an iPhone. Well, this spoofing is actually really effective because `localForage` will now emit an error if you configure it to exclusively use indexedDB.
> 
> If you want to have a moment of profound ennui and see indexedDB in action... just turn off Chrome dev tool's emulation so that you're actually using Chrome's own user-agent. It should work then.
> 
> You can read more info here: #254. TLDR;x2 indexedDB for iOS is not ready.

Found it via Google and need to add something so that the next person  who finds this has a nicer day, even if you switch your device in DevTools and reload it can happen that something hangs, and it keeps for example the user-agent you had before switching. That added some confusion for me because my device showed definitive it was not on IOS emulation but localForge still told me that the type was not available or I had funny switching where it sometimes used to or the other driver restarting chrome helped in the end after I read this threat and figured that it  was a dev  tool  bug.

**Comment by GentileFulvio** [comment]
> I'm using localForage in a cordova app and when I debug it with chrome & dev tools open (in device mode), with an iOS device selected, I get this error.
> 
> I have localForage configured to use a SQLite driver first and then fall back on IndexedDB. Then I tried adding WebSQL and localStorage; still had the same result. The dev tools will change the user agent string value when you switch between different devices to emulate and I'm assuming that's how this decision is being made.

This helped me A LOT! thanks
